### ISSUE TITLE
Status message not exposed to assistive technologies

### ISSUE THREAD

**Issue by karlgroves** [WordPress/gutenberg#15293]
Status message not exposed to assistive technologies

-   **Severity:** Medium
-   **Affected Populations:**
    -   Blind
    -   Low-Vision
    -   Cognitively Impaired
-   **Platform(s):**
    -   Windows - Screen Reader
    -   Mac - VoiceOver
    -   Android - TalkBack
    -   iOS - VoiceOver
-   **Components affected:**
    -   Edit Media

#### Issue description

Within the Edit Media page, when activating the "Restore Image"
button, a message is shown above the image while the Restore button
itself disappears.

Since the button would have been focused at the time when activated by
keyboard, this causes the keyboard focus position to be lost and reset
to the top of the page.

The message itself is also not announced by screen readers, and may not
be visible to screen magnification users if it appears outside their
current view.

##### Issue Code

```
    <div class="imgedit-panel-content wp-clearfix">


        <div class="error">


            <p>Cannot save image metadata.</p>


        </div>


        <div class="imgedit-menu wp-clearfix">


            ...


        </div>


    </div>





    ...





    <div class="imgedit-panel-content wp-clearfix">


        <div class="updated">


            <p>Image restored successfully.</p>


        </div>


        <div class="imgedit-menu wp-clearfix">


            ...


        </div>


    </div>
```

#### Remediation Guidance

Add `tabindex="-1"` to the message, and then programatically focus
it when it appears. This will maintain a logical focus position, and
ensure that all visual users will see it, while assistive technologies
announce it.

Adding the `alert` role will also help to convey to screen readers
that this is an alert-type message.

##### Recommended Code

```
    <div class="imgedit-panel-content wp-clearfix">


        <div class="error" tabindex="-1" role="alert">


            <p>Cannot save image metadata.</p>


        </div>


        <div class="imgedit-menu wp-clearfix">


            ...


        </div>


    </div>





    ...





    <div class="imgedit-panel-content wp-clearfix">


        <div class="updated" tabindex="-1" role="alert">


            <p>Image restored successfully.</p>


        </div>


        <div class="imgedit-menu wp-clearfix">


            ...


        </div>


    </div>
```

#### Relevant standards

-   [2.1.1 Keyboard
    (Level A)](https://www.w3.org/TR/WCAG20/#keyboard-operation-keyboard-operable)

**Note:** This issue may be a duplicate with other existing accessibility-related bugs in this project. This issue comes from the Gutenberg accessibility audit, performed by [Tenon](https://www.tenon.io) and funded by [WP Campus](https://wpcampus.org/). This issue is GUT-73 in Tenon's report

**Comment by afercia** [comment]
Worth noting the media modals aren't part of Gutenberg. This issue should be moved to the core Trac under the Media component.

**Comment by melchoyce** [comment]
Full report screenshot:
![image](https://user-images.githubusercontent.com/2846578/57186829-bc9dfc00-6e9a-11e9-99be-b0975c99aea6.png)

**Comment by anevins12** [comment]
Moved to Trac: https://core.trac.wordpress.org/attachment/ticket/47147

**Comment by afercia** [comment]
Closing as moved to Trac.
